chutney/server/src/main/java/com/chutneytesting/execution/domain/purge/PurgeServiceImpl.java

```diff
@@ -53,40 +56,75 @@ public class PurgeServiceImpl implements PurgeService {
+        CampaignRepository campaignRepository,
+        CampaignExecutionRepository campaignExecutionRepository,
+        int maxScenarioExecutionsConfiguration,
+        int maxCampaignExecutionsConfiguration
+    ) {
+        this(testCaseRepository,
+            executionsRepository,
+            campaignRepository,
+            campaignExecutionRepository,
+            maxScenarioExecutionsConfiguration,
+            0,
+            maxCampaignExecutionsConfiguration,
+            0);
+    }
+
     public PurgeServiceImpl(
         TestCaseRepository testCaseRepository,
         ExecutionHistoryRepository executionsRepository,
         CampaignRepository campaignRepository,
         CampaignExecutionRepository campaignExecutionRepository,
-        Integer maxScenarioExecutionsConfiguration,
-        Integer maxCampaignExecutionsConfiguration
+        int maxScenarioExecutionsConfiguration,
+        int scenarioBeforeMillisecondsTimeExecutionsConfiguration,
+        int maxCampaignExecutionsConfiguration,
+        int campaignsBeforeMillisecondsTimeExecutionsConfiguration
     ) {
-        Integer maxScenarioExecutions = validateConfigurationLimit(maxScenarioExecutionsConfiguration, "maxScenarioExecutions");
-        Integer maxCampaignExecutions = validateConfigurationLimit(maxCampaignExecutionsConfiguration, "maxCampaignExecutions");
+        int maxScenarioExecutions = validateConfigurationLimit(true, maxScenarioExecutionsConfiguration, "maxScenarioExecutions", 10);
+        int maxCampaignExecutions = validateConfigurationLimit(true, maxCampaignExecutionsConfiguration, "maxCampaignExecutions", 10);
+        int scenarioBeforeHoursTimeExecutions = validateConfigurationLimit(false, scenarioBeforeMillisecondsTimeExecutionsConfiguration, "scenarioBeforeMillisecondsTimeExecutions", 86400000);
+        int campaignsBeforeHoursTimeExecutions = validateConfigurationLimit(false, campaignsBeforeMillisecondsTimeExecutionsConfiguration, "campaignsBeforeMillisecondsTimeExecutions", 86400000);
 
-        this.scenarioPurgeService = buildScenarioService(testCaseRepository, executionsRepository, maxScenarioExecutions);
-        this.campaignPurgeService = buildCampaignService(campaignRepository, campaignExecutionRepository, maxCampaignExecutions);
+        this.scenarioPurgeService = buildScenarioService(testCaseRepository, executionsRepository, maxScenarioExecutions, scenarioBeforeHoursTimeExecutions);
+        this.campaignPurgeService = buildCampaignService(campaignRepository, campaignExecutionRepository, maxCampaignExecutions, campaignsBeforeHoursTimeExecutions);
     }
 
-    private static Integer validateConfigurationLimit(Integer configurationLimit, String configName) {
-        if (configurationLimit <= 0) {
-            LOGGER.warn("Purge configuration limit must be positive. Defaulting {} to {}", configName, 10);
-            return 10;
+    private static int validateConfigurationLimit(
```

Comment:
make two method to be more understandable

Reply:
or remove it 1ms == 0

Reply:
I chose to keep just a positive check, which is mandatory to exclude exception in purge process;